lib/client-manager.js

```diff
@@ -177,11 +187,10 @@ ClientManager.prototype.cacheAnnotations = function (pkgName, annotations, cb) {
     annotation.fingerprint = hash(JSON.stringify(annotation), '')
     return JSON.stringify(annotation)
   })
+  this.client.del(this.key(pkgName))
+  this.client.setex('lock_' + this.key(pkgName), failures ? this.ttlFailure : this.ttl, true)
```

Comment:
Seems like it would be better to keep the cached annotations (not delete them) until after the expiring lock has been set. Thoughts?